services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/team/EmailParseUtil.java

```diff
@@ -44,4 +46,32 @@ public static String getTokenFromString(String s, String startString, String end
 		return token;
 	}
 
+	/*
+	 * Returns the substring token in s that is representing the position tokenIndex in templatePieces
+	 * if s matches the template represented by templatePieces. Otherwise returns null.
+	 */
+	public static String getTokenFromString(String s, List<String> templatePieces, int tokenIndex) {
+		String regex = "";
+		for (int i = 0; i < templatePieces.size(); i++) {
+			if (i == tokenIndex) {
+				// Current piece is the matching target
+				regex += "(.*?)";
+			} else {
+				String piece = templatePieces.get(i);
+				if (piece.charAt(0) == '#' && piece.charAt(piece.length() - 1) == '#') {
+					// Current piece is a variable
+					regex += ".*";
+				} else {
+					// Current piece is a literal string
+					regex += Pattern.quote(piece);
+				}
+			}
+		}
+		Pattern pattern = Pattern.compile(regex);
```

Comment:
The use of this method is for testing EmailUtils.readMailTemplate(String, Map<String, String>). 
The above static method should be fully tested. Its caller should not test its logic.
Please make sure that EmailUtils.readMailTemplate() is fully tested, and remove the redundant tests.